src/Functions/byteSize.cpp

```diff
@@ -0,0 +1,281 @@
+        {
+            const ColumnNullable * col_null = checkAndGetColumn<ColumnNullable>(column);
+            for (size_t i = 0; i < vec_size; ++i)
+            {
+                UInt64 byte_size = sizeof(bool);
+                if (!col_null->isNullAt(i))
+                    byte_size += byteSizeForNestedItem(col_null->getNestedColumn(), i);
+                vec_res[i] += byte_size;
+            }
+            return true;
+        }
+        else if (which.isTuple()) // TypeIndex::Tuple
+        {
+            const DataTypeTuple * data_type_tuple = checkAndGetDataType<DataTypeTuple>(data_type);
+            const ColumnTuple * col_tuple = checkAndGetColumn<ColumnTuple>(column);
+
+            size_t tuple_size = col_tuple->tupleSize();
+            for (size_t col_idx = 0; col_idx < tuple_size; ++col_idx)
+            {
+                const IDataType * nested_data_type = data_type_tuple->getElements()[col_idx].get();
+                const IColumn * nested_col = col_tuple->getColumnPtr(col_idx).get();
+                byteSizeOne(nested_data_type, nested_col, vec_res);
+            }
+            return true;
+        }
+        else if (which.isLowCardinality()) // TypeIndex::LowCardinality
+        {
+            const ColumnLowCardinality * col_low = checkAndGetColumn<ColumnLowCardinality>(column);
+            size_t byte_size = col_low->getSizeOfIndexType();
+            for (size_t i = 0; i < vec_size; ++i)
+                vec_res[i] += byte_size;
+            return true;
+        }
+
+        return false;
+    }
+
+    static bool byteSizeForArrayByDataType(const ColumnArray * col_arr, ColumnUInt64::Container & vec_res)
+    {
+        TypeIndex type_id = col_arr->getData().getDataType();
+        UInt64 byte_size = 0;
+        if (byteSizeByTypeId(type_id, byte_size))
+        {
+            size_t vec_size = vec_res.size();
+            const auto & offsets = col_arr->getOffsets();
+            ColumnArray::Offset prev_offset = 0;
+            for (size_t i = 0; i < vec_size; ++i)
+            {
+                size_t array_size = offsets[i] - prev_offset;
+                vec_res[i] += array_size * byte_size + sizeof(offsets[0]);
+                prev_offset = offsets[i];
+            }
+            return true;
+        }
+        return false;
+    }
+
+    static inline bool byteSizeForConstArray(const IColumn * column, ColumnUInt64::Container & vec_res) {
+        const ColumnConst * col_arr = checkAndGetColumnConst<ColumnArray>(column);
+        if (!col_arr)
+            return false;
+        size_t vec_size = vec_res.size();
+        const UInt64 byte_size = col_arr->byteSize();
+        for (size_t i = 0; i < vec_size; ++i)
+            vec_res[i] += byte_size;
+        return true;
+    }
+
+    static UInt64 byteSizeForNestedItem(const IColumn & column, size_t idx)
+    {
+        if (const ColumnString * col_str = checkAndGetColumn<ColumnString>(&column))
```

Comment:
It's quite strange that only a few types are supported for array elements.
What about Array(Tuple(Nullable(FixedString), ...))?